primitives/header-chain/src/lib.rs

```diff
@@ -75,11 +75,11 @@ pub trait HeaderChain<H> {
 	/// Get the best authority set known to the header chain.
 	fn authority_set() -> AuthoritySet;
 
-	/// Write the given header to the underlying pallet storage.
+	/// Write a finalized chain of headers to the underlying pallet storage.
 	///
-	/// It is assumed that this header has been finalized.
-	#[allow(clippy::result_unit_err)]
-	fn import_header(header: H) -> Result<(), ()>;
+	/// It is assumed that each header in this chain been finalized, and that the given headers are
+	/// in order (e.g vec![header_1, header_2, ..., header_n]).
```

Comment:
I'd add a similar comment as in `import_header_unchecked` - the caller has to ensure that `header_1` is a child of current best finalized header or we should add this simply check to the function implementation.

Reply:
> ... or we should add this simply check to the function implementation

Yeah, I guess the overhead of the extra storage read isn't that big a deal if it prevents people from accidentally bricking their chain